docs.feldera.com/docs/pipelines/modifying.md

```diff
@@ -0,0 +1,436 @@
+associated with the current version of the Feldera platform. This new runtime may generate
+a slightly different query plan for the same SQL tables and views, which will require bootstrapping.
+The affected views will be recomputed from scratch and streamed to the output connectors even
+though their contents haven't changed.
+
+If this is undesirable, use one of the following methods to avoid updating the SQL runtime of a pipeline:
+
+1. Avoid updating pipeline's runtime. As long as you don't invoke `/update_runtime` or modify
+   the pipeline's code, it will continue running with its original runtime.
+
+2. Pin the pipeline to a specific runtime version using the [`runtime_version` property](/pipelines/lifecycle).
+
+### Caveat 2: Starting from an S3 checkpoint
+
+Normally, when restarting, a pipeline picks up the checkpoint it made while stopping, containing
+the pipeline's latest state. If the pipeline crashed or was [force-stopped](https://docs.feldera.com/api/stop-the-pipeline-asynchronously-by-updating-the-desired-state), it will pick up the
+last [periodic checkpoint](https://docs.feldera.com/pipelines/configuration/#program-configuration), if any.
+
+Alternatively, the user can start the pipeline from a checkpoint in an S3-compatible object
+store, created using the [S3 sync](https://docs.feldera.com/pipelines/checkpoint-sync) feature. This checkpoint may have been created by an older
+version of the same pipeline or even by a different pipeline.  This situation is handled in the
+same way: the pipeline computes a diff between the current and the checkpointed query plan, bootstraps
+any new and modified views and clears any tables that are not part of the checkpoint.
+
+### Caveat 3: Views that have not been modified may require bootstrapping
+
+The Feldera SQL compiler performs some global optimizations across multiple views. Modifying or
+removing one of the views can affect these optimizations, potentially changing the query plan
+for other views, not directly affected by the change.
+
+### Limitation 1: Tables must be materialized
+
+Bootstrapping relies on existing pipeline state to initialize new and modified views.
+[Materializing](/sql/materialized) all tables in the program by declaring them with `'materialized' = 'true'`
+ensures that the pipeline can be bootstrapped after arbitrary changes from this materialized
+state.
+
+A program where not all tables are materialized may fail to start from a checkpoint and require
+clearing the pipeline's state and restarting the pipeline from scratch instead.
+
+### Limitation 2: Bootstrapping does not work with `LATENESS`
+
+Programs with [`LATENESS` annotations](/tutorials/time-series) currently cannot be bootstrapped. These annotations are used
+by the query engine to discard state that is not used to evaluate any of the views in the program.
+However, the modified program may contain a different set of queries that require the discarded state.
```

Comment:
if a modified program contains new views that may depend on the discarded data, they cannot be computed.